### PR TITLE
[MRG] add `--distance-matrix` option to `sourmash compare`

### LINKED ISSUE #2196: support distance matrix output from `compare`?
`sourmash compare` outputs a similarity matrix. Many tools want the opposite, a distance matrix.

It would be relatively straightforward to support `--distance` (outputs 1-value). This would work for jaccard, containment, and ANI outputs, which would avoid any errors brought on by custom scripts transforming values.

from discussion with @nvpatin

### PR DESCRIPTION
**Pull request by ctb**
This PR addresses #2196 by adding a `--distance-matrix` flag to `compare`.

This flag simply subtracts every value in the matrix from 1, converting it from a similarity matrix to a distance matrix. Easy peasy!

Along the way, I decided to clean up the compare tests in `tests/test_sourmash.py`, which makes for a somewhat messier diff, sorry :(. Specifically, I
* renamed the compare tests to all have `def test_compare` at the beginning of the test fn;
* switched the tests from decorator `c` to test fixture `runtmp`;
* added comments;
* refactored some shared sig and matrix loading code out of a number of tests;

TODO:
- [x] confirm that plot code needs no changes

Fixes https://github.com/sourmash-bio/sourmash/issues/2196

### CODE REVIEW

**Comment by ctb** [comment]
ready for review @sourmash-bio/devs !

**Comment by taylorreiter** [comment]
did you confirm that the plot code needs no changes as marked as a todo item?

File: src/sourmash/cli/compare.py [hunk 1]
```diff
@@ -82,6 +80,18 @@ def subparser(subparsers):
     subparser.add_argument(
         '-p', '--processes', metavar='N', type=int, default=None,
         help='Number of processes to use to calculate similarity')
+    subparser.add_argument(
+        '--distance-matrix', action='store_true',
+        help='output a distance matrix, instead of a similarity matrix'
+    )
+    subparser.add_argument(
+        '--similarity-matrix', action='store_false',
```
Comment (taylorreiter) [inline src/sourmash/cli/compare.py:88]: this one isn't documented `command-line.md`. Is that ok?

File: src/sourmash/commands.py [hunk 2]
```diff
@@ -169,16 +169,24 @@ def compare(args):
         similarity = compare_all_pairs(siglist, args.ignore_abundance,
                                        n_jobs=args.processes, return_ani=return_ani)
 
+    # if distance matrix desired, switch to 1-similarity
+    if args.distance_matrix:
+        similarity = 1 - similarity
```
Comment (taylorreiter) [inline src/sourmash/commands.py]: well. that's fine. but it is deeply confusing that the distance matrix will be named similarity. I get why it is tho so i accept this chaos and will move on with my life.

**Review by taylorreiter — APPROVED** [review]
a couple of comments, otherwise LGTM.

**Comment by taylorreiter** [comment]
also why is the docs build failing? (sry if its a dumb question bc i haven't been keeping up with sourmash CI)

**Comment by ctb** [comment]
> also why is the docs build failing? (sry if its a dumb question bc i haven't been keeping up with sourmash CI)

readthedocs is ALWAYS failing. Usually because I haven't pinned some version or something 😆 

I'll take a look in a bit.

File: src/sourmash/commands.py (hunk 2 above)
Comment (ctb) [reply src/sourmash/commands.py]: nah, you were right. renamed to `matrix` in 6091d39a.

File: src/sourmash/cli/compare.py (hunk 1 above)
Comment (ctb) [reply src/sourmash/cli/compare.py:88]: I think so - it shows up in `--help`, and it's the default, so hopefully it's obvious.

(the command-line docs aren't comprehensive; they merely try to be useful :)

**Comment by ctb** [comment]
> did you confirm that the plot code needs no changes as marked as a todo item?

unfortunately, it does indeed need changes. but I think I will do as a separate issue.

(oddly, I thought I'd made this comment ...somewhere, but apparently forgot it. yay.)

**Comment by ctb** [comment]
readthedocs builds failing issue: https://github.com/sourmash-bio/sourmash/issues/2227

**Comment by ctb** [comment]
`plot` adjustment issue: https://github.com/sourmash-bio/sourmash/issues/2228

**Comment by ctb** [comment]
thanks for the review @taylorreiter - I'm going to wait to merge until https://github.com/sourmash-bio/sourmash/pull/2215 is in!

**Pull request merged**